FlexAPI.php

```diff
@@ -126,6 +126,31 @@ public static function navigateTo($url) {
         die;
     }
 
+    public static function buildUrl($config) {
+        $config = setFieldDefault($config, 'scheme', FlexAPI::get('defaultUrlScheme'));
+        $queryString = '';
+        if (array_key_exists('queries', $config)) {
+            $pairs = [];
+            foreach($config['queries'] as $name => $value) {
+                array_push($pairs, "$name=".urlencode($value));
+            }
+            $queryString = '?'.implode('&', $pairs);
+        }
+        $path = sprintf('/%s/%s/%s',
+            FlexAPI::get('basePath'),
+            FlexAPI::get('apiPath'),
+            $config['endpoint']
+        );
+        $path = preg_replace('/[\/]+/', '/', $path);
+        return sprintf("%s://%s:%s%s%s",
+            $config['scheme'],
```

Comment:
Ich fände es hier besser $_SERVER['HTTPS'] auszuwerten, statt das Schema aus der Config zu nehmen. Letztenendlich würde ich aber gerne die Token-Url ganz aus der Config nehmen, und das Frontend statt das Backend addressieren. Da kann es dann noch eine "Bist du sicher?" Abfrage geben. 

Deine Lösung berücksichtigt nicht, dass es z.B. E-Mail-Virenscann Lösungen gibt, die prüfen ob hinter einer URL in einer E-Mail evtl. ein Trojaner lauert. Das ist dann ein Einfallstor: Der Angreifer sagt: Ich hab das Passwort vergessen und gibt ein neues vor, der Virenscanner bestätigt die Änderung...

Reply:
Ok, ich hab den Punkt nicht ganz verstanden. 

Wie sollen wir es mit Passwortänderung machen? Vlt. zufälliges Passwort generieren und dieses per Mail verschicken?

Reply:
Passwortänderung würde ich folgendes Verfahren vorschlagen:

1.  User sagt das er sein Passwort vergessen hat -> EMail an API
2.  Backend schickt eine URL zurück. Diese wird wie folgt ermittelt:
frontend_base_url (aus Konfig Datei) + '#/verify/lostPassword/'+token
3. Der User liest die E-Mail und ruft das Frontend auf
4. Das Frontend zeigt eine Meldung an, die der User mit JA bestätigen muss
5. Das Frontend ruft das Backend auf (API Call ist in der README.md dokumentiert)
6. Das Backend schickt dem User ein neues Passwort per E-Mail


Für Das bestätigen einer 
* E-Mail Änderung analog: Nur das die Frontend URL: frontend_base_url (aus Konfig Datei) + '#/verify/mailAddressChange/'+token aufruft (und das Passwort nicht geändert wird)
* Registrierung: frontend_base_url (aus Konfig Datei) + '#/verify/register/'+token

Hoffe das ist jetzt klarer. Sonst frag gerne nach.

Reply:
Ist jetzt aber nicht so wichtig, das in diesem Pull-Request abzuarbeiten.